exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/flatten/FlattenRecordBatch.java

```diff
@@ -263,6 +260,8 @@ protected IterOutcome doWork() {
 
     flattenMemoryManager.updateOutgoingStats(outputRecords);
 
+    logger.debug("BATCH_STATS, outgoing:\n {}", new RecordBatchSizer(this));
```

Comment:
Every execution (even when logging does not happen; i.e., above "debug") does invoke the call "new RecordBatchSizer()", which adds time/memory overhead (and GC to remove these objects).
   Instead -- better surround this logger.debug() call with ``` if (logger.isDebugEnabled()) { ``` 
see similar examples elsewhere (e.g. Foreman.java)

Reply:
@Ben-Zvi Good point. Thank you. I fixed that. Please take a look.